interfaces/cython/cantera/cti2yaml.py

```diff
@@ -0,0 +1,1648 @@
+            the functions `enable_motz_wise` or `disable_motz_wise`).
+        """
+        self.motz_wise = kwargs.pop('motz_wise', None)
+        Arrhenius.__init__(self, *args, **kwargs)
+
+
+class reaction(object):
+    """
+    A homogeneous chemical reaction with pressure-independent rate coefficient
+    and mass-action kinetics.
+    """
+    def __init__(self, equation, kf, id='', order='', options=[]):
+        r"""
+        :param equation:
+            A string specifying the chemical equation.
+        :param kf:
+            The rate coefficient for the forward direction. If a sequence of
+            three numbers is given, these will be interpreted as [A, b, E] in
+            the modified Arrhenius function :math:`A T^b exp(-E/\hat{R}T)`.
+        :param id:
+            An optional identification string.
+        :param order:
+            Override the default reaction orders implied by the reactant
+            stoichiometric coefficients. Given as a string of key:value pairs,
+            e.g., ``"CH4:0.25 O2:1.5"``.
+        :param options: Processing options, as described in
+            `Options <https://cantera.org/tutorials/cti/reactions.html#options>`_.
+            May be one or more (as a list) of the
+            following: ``'duplicate'``, ``'negative_A'``,`` 'negative_orders'``,
+            ``'nonreactant_orders'``.
+        """
+        self._e = equation
+        self._order = getComposition(order)
+        self._num = len(_reactions['reactions']) + 1
+        self._id = id
+        self._options = [options] if isinstance(options, str) else options
+        self._kf = Arrhenius(*kf) if isinstance(kf, (list, tuple)) else kf
+        self._type = 'elementary'
+        _reactions['reactions'].append(self)
+
+    @classmethod
+    def to_yaml(cls, representer, node):
+        out = BlockMap()
+        node.get_yaml(out)
+        return representer.represent_dict(out)
+
+    def get_yaml(self, out):
+        out['equation'] = self._e
+        out.yaml_add_eol_comment('Reaction {}'.format(self._num), 'equation')
+        if self._type not in ('elementary', 'edge', 'surface'):
+            out['type'] = self._type
+
+        if self._id:
+            out['id'] = self._id
+
+        if self._type in ('elementary', 'three-body', 'edge', 'surface'):
+            out['rate-constant'] = self._kf
+
+        if 'duplicate' in self._options:
+            out['duplicate'] = True
+        if 'negative_A' in self._options:
+            out['negative-A'] = True
+
+        if self._order:
+            out['orders'] = FlowMap(self._order.items())
+        if 'negative_orders' in self._options:
+            out['negative-orders'] = True
+        if 'nonreactant_orders' in self._options:
+            out['nonreactant-orders'] = True
+
+
+class three_body_reaction(reaction):
+    """
+    A three-body reaction.
+    """
+    def __init__(self, equation, kf, efficiencies='', id='', options=[]):
```

Comment:
That would break any existing input files which specify `options` as a positional argument, where it would be misinterpreted as the `order` in the parent class.

Reply:
Dunno how I missed that...